src/Microsoft.DotNet.Interactive.Formatting/Formatter.cs

```diff
@@ -766,4 +685,40 @@ private static bool TryGetPropertyValue<T>(object fromObject, string propertyNam
         value = default;
         return false;
     }
+
+    internal static bool ShouldIncludePropertiesInOutput(this Type type)
+    {
+        if (type.IsArray)
+        {
+            return false;
+        }
+
+        if (type.IsNestedPrivate) //e.g. RangeIterator
+        {
+            return false;
+        }
+
+        if (typeof(ICollection).IsAssignableFrom(type))
```

Comment:
Curious why properties are included for types that implement `IEnumerable` but not `ICollection`.

Reply:
`ICollection` is a lot more specific and includes a `Count` property which is redundant with the existing sequence formatters, which show e.g. `(13 more)` after hitting the list expansion limit.